AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -2165,11 +2168,38 @@ public static void clearUserContent() {
   }
 
   private static void parseSdkCounters(JSONObject response) {
+    parseSdkCounters(response, CountAggregator.INSTANCE);
+  }
+
+  @VisibleForTesting
+  public static void parseSdkCounters(JSONObject response, CountAggregator countAggregator) {
     JSONArray enabledCounters = response.optJSONArray(
             Constants.Keys.ENABLED_COUNTERS);
     if (enabledCounters != null) {
-      HashSet counterSet = new HashSet<>(Arrays.asList(enabledCounters));
-      CountAggregator.INSTANCE.setEnabledCounters(counterSet);
+      HashSet<String> counterSet = new HashSet<>(toList(enabledCounters));
+      countAggregator.setEnabledCounters(counterSet);
+    }
+  }
+
+  private static void parseFeatureFlags(JSONObject response) {
+    parseFeatureFlags(response, FeatureFlagManager.INSTANCE);
+  }
+
+  @VisibleForTesting
+  public static void parseFeatureFlags(JSONObject response, FeatureFlagManager featureFlagManager) {
```

Comment:
currently this method is parsing feature flags AND setting them on the featureFlagManager. Instead, lets have this method parse the feature flags and return a List<String>. Then, we can test this method easily, and set that list to the enabled near line 864.

Reply:
counters and flags need to be set above the line 864 block or else shouldParse... tests would fail, I set them in lines 824 and 826 instead, similar to iOS structure